Cabal/Cabal.cabal

```diff
@@ -410,6 +480,9 @@ library
     Distribution.Types.ComponentRequestedSpec
     Distribution.Types.TargetInfo
     Distribution.Utils.NubList
+    Distribution.Utils.UnionFind
+    Distribution.Utils.Base62
+    Distribution.Utils.Progress
```

Comment:
Yeah, a lot of these actually are used by cabal-install. For example, in the Backpack namespace, cabal-install imports Distribution.Backpack.ConfiguredComponent, Distribution.Backpack.LinkedComponent, Distribution.Backpack.ComponentsGraph, Distribution.Backpack.ModuleShape, Distribution.Backpack.ModSubst, and Distribution.Backpack. So I guess we could hide the stuff for actually implementing mix-in linking, but a good chunk of this stuff also needs to stay.